go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -144,70 +144,83 @@ func ToK8sPodSpec(ctx context.Context, tCtx pluginsCore.TaskExecutionContext) (*
 	return pod, nil
 }
 
-func BuildPodWithSpec(podTemplate *v1.PodTemplate, podSpec *v1.PodSpec, primaryContainerName string) (*v1.Pod, error) {
-	pod := v1.Pod{
-		TypeMeta: v12.TypeMeta{
-			Kind:       PodKind,
-			APIVersion: v1.SchemeGroupVersion.String(),
-		},
+func MergePodSpecs(podTemplatePodSpec *v1.PodSpec, podSpec *v1.PodSpec, primaryContainerName string) (*v1.PodSpec, error) {
```

Comment:
Do we need to add any specific `if podTemplatePodSpec == nil` or `if podSpec == nil` functionality here now that it is a separate function? I don't recall how `mergo.Merge` operates on nil data.